.github/workflows/build-ci.sh

```diff
@@ -58,65 +103,81 @@ if [[ $setup_python_venv -eq 1 ]]; then
     fi
 
     pip install --upgrade pip
-    pip install ninja-build
     pip install torch torchvision torchaudio --index-url $torch_source
     pip install pybind11
-  else
-    source "$py_venv_path/bin/activate"
+    pip install build
   fi
 else
   echo "Skipping Python venv setup"
   echo "Make sure to have a correct Python environment set up"
 fi
 
 if [[ $checkout_and_build_llvm -eq 1 ]]; then
+  reconfigure_llvm=0
   if [ ! -d "$llvm_path" ]; then
-    git clone https://github.com/oowekyala/llvm-project "$llvm_path"
+    git clone https://github.com/llvm/llvm-project "$llvm_path"
+    
+    reconfigure_llvm=1
+  fi
+
+  cd "$llvm_path"
+
+  if [ $reconfigure -eq 1 ] || [ $reconfigure_llvm -eq 1 ]; then
+    git reset --hard
+    git fetch
+    git checkout llvmorg-19.1.3
 
-    cd "$llvm_path"
+    patch_dir="$project_root/patches/llvm"
+    for patch in $(ls $patch_dir); do
+      git apply $patch_dir/$patch
+    done
 
-    git checkout cinnamon-llvm
-    cmake -S llvm -B build -GNinja \
+    cmake -S llvm -B build \
       -DLLVM_ENABLE_PROJECTS="mlir;llvm;clang" \
       -DLLVM_TARGETS_TO_BUILD="host" \
       -DLLVM_ENABLE_ASSERTIONS=ON \
       -DMLIR_ENABLE_BINDINGS_PYTHON=ON \
-      -DLLVM_BUILD_TOOLS=OFF \
       -DCMAKE_BUILD_TYPE=Release \
       -DBUILD_SHARED_LIBS=ON \
+      -DLLVM_INCLUDE_TESTS=OFF \
+      -DLLVM_INCLUDE_BENCHMARKS=OFF \
       -DLLVM_OPTIMIZED_TABLEGEN=ON \
       -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=SPIRV \
       $LLVM_CMAKE_OPTIONS
   fi
 
-  cd "$llvm_path"
-  git pull
   cmake --build build --target all llc opt
 
   export PATH=$llvm_path/build/bin:$PATH
 else
   echo "Skipping LLVM checkout and build"
-fi
-
-llvm_dir=${LLVM_BUILD_DIR:-"$llvm_path/build"}
-echo "Using LLVM installation in $llvm_dir"
-if [[ ! -d "$llvm_dir" ]]; then
-  echo "No LLVM installation found"
-  exit 1
+  echo "The following steps will need LLVM_DIR and MLIR_DIR to be set in their respective <STEP>_CMAKE_OPTIONS"
 fi
 
 if [[ $checkout_and_build_torch_mlir -eq 1 ]]; then
+  reconfigure_torch_mlir=0
   if [ ! -d "$torch_mlir_path" ]; then
     git clone https://github.com/llvm/torch-mlir "$torch_mlir_path"
+
+    reconfigure_torch_mlir=1
   fi
+
   cd "$torch_mlir_path"
-  if [ ! -d "build" ]; then
-    mkdir build
 
-    git checkout snapshot-20240127.1096
-    cmake -S . -B build -GNinja \
-      -DLLVM_DIR="$llvm_dir/lib/cmake/llvm" \
-      -DMLIR_DIR="$llvm_dir/lib/cmake/mlir" \
+  if [ $reconfigure -eq 1 ] || [ $reconfigure_torch_mlir -eq 1 ]; then
+    git reset --hard
+    git fetch
```

Comment:
The git reset and fetch are already removed in some commits on another fork branch as they caused issues with caching in the CI runner. I will add the relevant commits to this PR. I will also have a look at shallow cloning only the needed revision. This will probably cause issues when we bump the LLVM version in the future, but simply deleting the directory and letting the script re-clone it should be an easy enough solution to this.

Reply:
Suggestions implemented in 5748884.